coreax/kernel.py

```diff
@@ -391,7 +390,7 @@ def slice_sum(accumulated_sum: Array, y_block: Data) -> tuple[Array, Array]:
 
 
 def _block_data_convert(
-    x: Union[ArrayLike, Data], block_size: Union[int, None]
+    x: Union[ArrayLike, Data], block_size: Optional[int]
```

Comment:
While it isn't obvious or clear from https://docs.python.org/3/library/typing.html#typing.Optional, I believe the only time `Optional` should be used is if we have a default assignment of `None`. In all other cases, like here, I think `Union[int, None]` is more appropriate. Otherwise, the current type hint reads as if `block_size` is optional, even though it is not. I think this is why the `int | None` syntax, supported in newer versions of python, is better (there is no ambiguity).